test/e2e.test.ts

```diff
@@ -37,4 +37,34 @@
       )
     )
   })
+
+  it('expect no callback loop', async () => {
+    
+    const page = await createPage()
+    const resources: string[] = []
+    const logs: string[] = []
+
+    page.on('request', res => {
+      resources.push(res.url())
+    })
+    page.on('console', event => {
+      logs.push(event.text())
+    })
+
+    await page.goto(url('/'), { waitUntil: 'networkidle' })
+    expect(resources.every(r => !r.includes('challenges.cloudflare.com')))
```

Comment:
uhh... Can we ignore url sanitization in tests files ?

Reply:
Well it's not a security risk to the user of course, but I think it would be good practice to apply the code style recommendation (under "Show more details" -> "show more") to have code that follows good practice. I can do that later :raised_hands:

Reply:
Are the lines the alert triggers on even required? That check is already part of the test above it, appears to me.